pkg/ephemeral-disk/ephemeral-disk.go

```diff
@@ -0,0 +1,152 @@
+package ephemeraldisk
+
+import (
+	"fmt"
+	"path/filepath"
+
+	kubev1 "k8s.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+)
+
+const (
+	filePrefix              = "ephemeral-disk-image"
+	fileType                = "qcow2"
+	ephemeralDiskOwner      = "qemu"
+	mountBaseDir            = "/var/run/ephemeraldisk"
+	ephemeralContainerImage = "kubevirt/ephemeral-disk:devel"
+)
+
+func generateVMBaseDir(vm *v1.VirtualMachine) string {
+	return fmt.Sprintf("%s", mountBaseDir)
+}
+func GenerateVolumeMountDir(vm *v1.VirtualMachine, volumeName string) string {
+	baseDir := generateVMBaseDir(vm)
+	return fmt.Sprintf("%s/%s", baseDir, volumeName)
+}
+
+func GetFilePath(vm *v1.VirtualMachine, volumeName string) (string, string, error) {
+	volumeMountDir := GenerateVolumeMountDir(vm, volumeName)
+	return volumeMountDir + "/" + "disk.qcow2", "qcow2", nil
+}
+
+func GenerateContainers(vm *v1.VirtualMachine) ([]kubev1.Container, error) {
+	var containers []kubev1.Container
+
+	initialDelaySeconds := 2
+	timeoutSeconds := 5
+	periodSeconds := 5
+	successThreshold := 2
+	failureThreshold := 5
+
+	// Make VM Image Wrapper Containers
+	for _, volume := range vm.Spec.Volumes {
```

Comment:
Afaik all that is not necessary. kubevirt/libvirt already containes `qemu-img`. Just create an empty qcow2 file in the virt-launcher container and place it in the empty-dir directory for e.g. registry disks in a unique location. We would then follow the same pattern like the cloud-init volume (the whole disk content is predictable for virt-launcher).